ddcommon/src/tracer_metadata.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2023-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+/// This struct MUST be backward compatible.
+#[derive(serde::Serialize, Debug)]
+#[repr(C)]
+pub struct TracerMetadata {
+    /// Version of the schema.
+    pub schema_version: u8,
+    /// Runtime UUID.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub runtime_id: Option<String>,
+    /// Programming language of the tracer library (e.g., “python”). Refers to telemetry
+    /// for valid values.
+    pub tracer_language: String,
+    /// Version of the tracer (e.g., "1.0.0").
+    pub tracer_version: String,
+    /// Identifier of the machine running the process.
+    pub hostname: String,
+    /// Name of the service being instrumented.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub service_name: Option<String>,
+    /// Environment of the service being instrumented.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub service_env: Option<String>,
+    /// Version of the service being instrumented.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub service_version: Option<String>,
+}
+
+pub enum AnonymousFileHandle {
+    #[cfg(target_os = "linux")]
+    Linux(Box<memfd::Memfd>),
```

Comment:
Why is this boxed?

Reply:
addressed in 3604d921baada7ccbc66b6e7b9f866d01d7e4cbf